test/src/test/java/org/corfudb/runtime/collections/CorfuMapTest.java

```diff
@@ -0,0 +1,763 @@
+                .isNull();
+        assertThat(testMap.put("a", "b"))
+                .isEqualTo("a");
+        assertThat(testMap.get("a"))
+                .isEqualTo("b");
+        getRuntime().getObjectsView().TXEnd();
+        assertThat(testMap.get("a"))
+                .isEqualTo("b");
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void objectViewCorrectlyReportsInsideTX()
+            throws Exception {
+        assertThat(getRuntime().getObjectsView().TXActive())
+                .isFalse();
+        getRuntime().getObjectsView().TXBegin();
+        assertThat(getRuntime().getObjectsView().TXActive())
+                .isTrue();
+        getRuntime().getObjectsView().TXEnd();
+        assertThat(getRuntime().getObjectsView().TXActive())
+                .isFalse();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void canUpdateSingleObjectTransacationallyWhenCached()
+            throws Exception {
+        r.setCacheDisabled(false);
+
+        Map<String, String> testMap = getRuntime().getObjectsView()
+                .build()
+                .setStreamName("test")
+                .setTypeToken(CorfuTable.<String,String>getMapType())
+                .open();
+
+        getRuntime().getObjectsView().TXBegin();
+        assertThat(testMap.put("a", "a"))
+                .isNull();
+        assertThat(testMap.put("a", "b"))
+                .isEqualTo("a");
+        assertThat(testMap.get("a"))
+                .isEqualTo("b");
+        getRuntime().getObjectsView().TXEnd();
+        assertThat(testMap.get("a"))
+                .isEqualTo("b");
+    }
+
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void abortedTransactionsCannotBeReadOnSingleObject()
+            throws Exception {
+        Map<String, String> testMap = getRuntime().getObjectsView()
+                .build()
+                .setStreamName("test")
+                .setTypeToken(CorfuTable.<String,String>getMapType())
+                .open();
+
+        testMap.clear();
+        testMap.put("z", "z");
+        assertThat(testMap.size())
+                .isEqualTo(1);
+
+
+        getRuntime().getObjectsView().TXBegin();
+        assertThat(testMap.put("a", "a"))
+                .isNull();
+        assertThat(testMap.put("a", "b"))
+                .isEqualTo("a");
+        assertThat(testMap.get("a"))
+                .isEqualTo("b");
+        testMap.clear();
+        getRuntime().getObjectsView().TXAbort();
+        assertThat(testMap.size())
+                .isEqualTo(1);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void modificationDuringTransactionCausesAbort()
+            throws Exception {
+        Map<String, String> testMap = getRuntime().getObjectsView()
+                .build()
+                .setStreamName("A")
+                .setTypeToken(CorfuTable.<String,String>getMapType())
+                .open();
+
+        assertThat(testMap.put("a", "z"));
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Complete the assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2970)